arbnode/dataposter/storage/storage.go

```diff
@@ -24,3 +28,125 @@ type QueuedTransaction struct {
 	Created         time.Time // may be earlier than the tx was given to the tx poster
 	NextReplacement time.Time
 }
+
+// LegacyQueuedTransaction is used for backwards compatibility.
+// Before https://github.com/OffchainLabs/nitro/pull/1773: the queuedTransaction
+// looked like this and was rlp encoded directly. After the pr, we are store
+// rlp encoding of Meta into queuedTransaction and rlp encoding it once more
+// to store it.
+type LegacyQueuedTransaction struct {
+	FullTx          *types.Transaction `rlp:"nil"`
+	Data            types.DynamicFeeTx
+	Meta            BatchPosterPosition
+	Sent            bool
+	Created         time.Time // may be earlier than the tx was given to the tx poster
+	NextReplacement time.Time
+}
+
+// This is also for legacy reason. Since Batchposter is in arbnode package,
+// we can't refer to BatchPosterPosition type there even if we export it (that
+// would create cyclic dependency).
+// Ideally we'll factor out Batch Poster from arbnode into separate package
+// and BatchPosterPosition into another separate package as well.
+// For the sake of minimal refactoring, that struct is duplicated here.
+type BatchPosterPosition struct {
```

Comment:
No action needed here, just noting that we'll just drop this struct in a few releases anyways when we drop legacy encoding so this problem will fix itself :)

Reply:
Added that comment.